x/pep/keeper/msg_server_get_private_keyshares.go

```diff
@@ -0,0 +1,129 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	sdkerrors "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	commontypes "github.com/Fairblock/fairyring/x/common/types"
+	kstypes "github.com/Fairblock/fairyring/x/keyshare/types"
+	"github.com/Fairblock/fairyring/x/pep/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+)
+
+func (k msgServer) GetPrivateKeyshares(goCtx context.Context, msg *types.MsgGetPrivateKeyshares) (*types.MsgGetPrivateKeysharesResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	entry, found := k.GetPrivateRequest(ctx, msg.ReqId)
+	if !found {
+		pubkey, found := k.GetActivePubKey(ctx)
+		if !found {
+			return &types.MsgGetPrivateKeysharesResponse{}, errors.New("entry and pubkey not found")
+		}
+
+		entry.Creator = ""
+		entry.Amount = sdk.NewCoin("ufairy", math.NewInt(0))
+		entry.EncryptedKeyshares = make([]*commontypes.EncryptedKeyshare, 0)
+		entry.Pubkey = pubkey.PublicKey
+		entry.ReqId = msg.ReqId
+
+		k.SetPrivateRequest(ctx, entry)
+	} else {
+		if !entry.Amount.IsNil() || !entry.Amount.IsZero() {
+			requester := sdk.MustAccAddressFromBech32(msg.Creator)
+			creator := sdk.MustAccAddressFromBech32(entry.Creator)
+			err := k.bankKeeper.SendCoins(ctx, requester, creator, sdk.NewCoins(entry.Amount))
```

Comment:
Sorry I don't quite get this one, isn't the creator for `RequestPrivateKeyShare` & `GetPrivateKeyShare` usually the same guy just like in the updated `pep.sh` script ?

Reply:
removed

Reply:
In the updated code, I think we should handle the error from the `SendCoinsFromAccountToModule()`

Reply:
added check